libs/src/services/discoveryProvider/index.js

```diff
@@ -547,8 +558,34 @@ class DiscoveryProvider {
 
     try {
       const response = await axios(axiosRequest)
-      return Utils.parseDataFromResponse(response)['data']
+      const parsedResponse = Utils.parseDataFromResponse(response)
+
+      if (
+        'latest_indexed_block' in parsedResponse &&
+        'latest_chain_block' in parsedResponse
+      ) {
+        const {
+          latest_indexed_block: indexedBlock,
+          latest_chain_block: chainBlock
+        } = parsedResponse
+
+        if (
+          !chainBlock ||
+          !indexedBlock ||
+          (chainBlock - indexedBlock) > UNHEALTHY_BLOCK_DIFF
+        ) {
+          // Clear any cached discprov
+          localStorage.removeItem(DISCOVERY_PROVIDER_TIMESTAMP)
+          // Select a new one
+          const endpoint = await this.autoSelectEndpoint()
```

Comment:
lets not do this now but rather than manually touching localStorage in this file, we just pass a force=true flag to autoSelectendpoint that does that?

Reply:
yeah good idea